package.py

```diff
@@ -0,0 +1,911 @@
+#!/usr/bin/env python3
+import argparse
+import base64
+import logging
+import os
+import platform
+import shutil
+import stat
+import subprocess
+from pathlib import Path
+from typing import Any, Callable, Literal, Optional
+from zipfile import ZipFile
+
+import requests
+
+from packaging import version
+
+pyinstaller_version = os.environ.get('PYINSTALLER_VERSION', '5.1')
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s: %(message)s',
+)
+logger = logging.getLogger('package')
+
+
+def log_group(name: str) -> Callable:
+    def start_group(group_name: str) -> None:
+        if os.environ.get('CI'):
+            subprocess.call(f'echo ::group::"{group_name}"', shell=True)
+        else:
+            logger.info(f'\n\n-----{group_name}-----\n\n')
+
+    def end_group() -> None:
+        if os.environ.get('CI'):
+            subprocess.call('echo ::endgroup::', shell=True)
+        else:
+            logger.info('\n\n-----------------\n\n')
+
+    def decorate(fn: Callable) -> Callable:
+        def wrapper(*args: Any, **kwargs: Optional[Any]) -> None:
+            start_group(name)
+            fn(*args, **kwargs)
+            end_group()
+
+        return wrapper
+    return decorate
+
+
+class Downloader:
+    @staticmethod
+    def download(url: str) -> str:
```

Comment:
Minor: Since you have the name hardcoded, why not use is at argument?
```suggestion
    def download(url: str, filename: str) -> str:
```

Reply:
It is not actually hard coded, it is taken from the URL.